htdocs/core/lib/modulebuilder.lib.php

```diff
@@ -411,7 +411,7 @@ function rebuildObjectSql($destdir, $module, $objectname, $newmask, $readdir = '
 		foreach ($object->fields as $key => $val) {
 			$i++;
 			if (!empty($val['index'])) {
-				$texttoinsert .= "ALTER TABLE llx_".strtolower($module).'_'.strtolower($objectname)." ADD INDEX idx_".strtolower($module).'_'.strtolower($objectname)."_".$key." (".$key.");";
+				$texttoinsert .= "ALTER TABLE llx_" . strtolower($module) . '_' . strtolower($objectname) . " ADD " . ($key == 'ref' ? "UNIQUE " : "") .  "INDEX idx_" . strtolower($module) . '_' . strtolower($objectname) . "_" . $key . " (" . $key . ($key == 'ref' && array_key_exists('entity', $object->fields) ? ", entity" : "") . ");";
```

Comment:
Can you also rename the idx into uk ? when unique key we use a different prefix for name of indexes.

Reply:
@eldy ok it is done.